src/coreclr/vm/amd64/JitHelpers_FastMP.asm

```diff
@@ -0,0 +1,75 @@
+; Licensed to the .NET Foundation under one or more agreements.
+; The .NET Foundation licenses this file to you under the MIT license.
+
+; ***********************************************************************
+; File: JitHelpers_InlineGetThread.asm, see history in jithelp.asm
+;
+; ***********************************************************************
+
+include AsmMacros.inc
+include asmconstants.inc
+
+CopyValueClassUnchecked equ     ?CopyValueClassUnchecked@@YAXPEAX0PEAVMethodTable@@@Z
+JIT_Box                 equ     ?JIT_Box@@YAPEAVObject@@PEAUCORINFO_CLASS_STRUCT_@@PEAX@Z
+
+extern CopyValueClassUnchecked:proc
+extern JIT_Box:proc
+
+; HCIMPL2(Object*, JIT_Box, CORINFO_CLASS_HANDLE type, void* unboxedData)
```

Comment:
It would be nice to get rid of this asm implementation and replace it with the portable C++ one. (It can be done in separate change.)

Reply:
I was having trouble getting the exact same assembly out of this one with a C++ implementation, so I didn't want to do it in this PR. I'll move it over in another PR and we can review the assembly differences there.

Reply:
The box helper is not super perf critical. The JIT will inline for all cases that matter. I think it would be fine to have something close enough - anything that does raise HELPER_METHOD_FRAME on a hot path should be ok.